iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParserBaseVisitorTest.java

```diff
@@ -590,6 +592,38 @@ void shouldParseKeyEmptyValue() {
       "visitEol <EOF>");
   }
 
+  @ParameterizedTest
+  @CsvSource({"foo#,foo#",
+    "foo#=,foo#",
+    "foo!,foo!",
+    "foo!=,foo!"
+  })
+  void shouldParseKeyThatContainsCommentIndicator(String code, String key) {
+    parseProperties(code);
+
+    assertThat(visitor.visited()).containsExactly(
+      "visitPropertiesFile %s<EOF><EOF>".formatted(code),
+      "visitRow %s<EOF>".formatted(code),
+      "visitLine %s<EOF>".formatted(code),
+      "visitKey %s".formatted(key),
+      "visitEol <EOF>");
+  }
+
+  @Test
+  void shouldParseKeyThatContainsHashAndEmptyValue() {
```

Comment:
Is the name correct? The value is not empty here

Reply:
Sure, I will rename it.